cache/Cache.js

```diff
@@ -0,0 +1,63 @@
+const EventEmitter = require('events')
+const _createRedisClient = require('./util/createRedisClient')
+const msgpack = require('msgpack-lite')
+
+// default entry expiration time in seconds
+const DEFAULT_EXPIRATION = 86400
+
+class Cache extends EventEmitter {
+  constructor ({ logger, nodes, defaultTtl, redisClientOptions }) {
+    super()
+    this._logger = logger
+    this._defaultTtl = defaultTtl || DEFAULT_EXPIRATION
+
+    this._onReady = () => this.emit('ready')
+    this._onError = (err) => this.emit('error', err)
+    this._onReconnecting = () => this.emit('reconnecting')
+    this._onNodeError = (err) => this.emit('node error', err)
+
+    const redisClient = this._redisClient =
+      _createRedisClient(nodes, redisClientOptions)
+
+    // forward events to clients so that they
+    // can handle situations appropriately
+    redisClient
+      .on('ready', this._onReady)
+      .on('reconnecting', this._onReconnecting)
+      .on('node error', this._onNodeError)
+      .on('error', this._onError)
+  }
+
+  async get (key) {
+    const encoded = await this._redisClient.getBuffer(key)
+    return encoded ? msgpack.decode(encoded) : encoded
+  }
+
+  async set (key, value, ttl) {
+    const data = msgpack.encode(value)
+    return this._redisClient.setex(key, ttl || this._defaultTtl, data)
+  }
+
+  async getEntity (key, ModelType) {
+    const data = await this.get(key)
+    return data ? ModelType.wrap(data) : data
+  }
+
+  async setEntity (key, model, ttl) {
+    const value = model.clean()
+    return this.set(key, value, ttl)
+  }
+
+  close () {
+    // remove all listeners
+    this._redisClient
+      .removeListener('ready', this._onReady)
+      .removeListener('reconnecting', this._onReconnecting)
+      .removeListener('node error', this._onNodeError)
+      .removeListener('error', this._onError)
+
+    this._redisClient.disconnect()
```

Comment:
I switched to `quit`. I'm not sure how to ensure that we don't indefinitely wait aside from adding a timeout. If you would like, I could add the timeout, but for now I think this is sufficient.